pkg/auth/auth.go

```diff
@@ -0,0 +1,162 @@
+package auth
+
+import (
+	sha "crypto/sha512"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/casbin/casbin/v2/model"
+)
+
+type authRecord struct {
+	role   string
+	expiry time.Time
+}
+
+type apiKeys map[string]authRecord
+
+type Authenticator struct {
+	user, pass string
+	apiKeys    apiKeys
+	enforcer   *casbin.Enforcer
+}
+
+func New() (*Authenticator, error) {
+	m, err := model.NewModelFromString(`
+	[request_definition]
+	r = sub, obj, act
+	
+	[policy_definition]
+	p = sub, obj, act
+	
+	[policy_effect]
+	e = some(where (p.eft == allow))
+	
+	[matchers]
+	m = r.sub == p.sub && keyMatch(r.obj, p.obj) && regexMatch(r.act, p.act)`)
+
+	if err != nil {
+		return nil, err
+	}
+
+	e, err := casbin.NewEnforcer(m)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := applyPolicies(e); err != nil {
+		return nil, err
+	}
+
+	auth := Authenticator{
+		user:     os.Getenv("AUTH_USERNAME"),
```

Comment:
The New constructor should receive variables for user and pass as arguments, instead to put a hard requirement on env variable configuration. The configuration should be handled in the cmd package and passed to other components as variables. With this approach, configuration is leaking into components. Another thing is that variables are names pretty generically. This can cause a miss-configuration, I would suppose to prefix them with BEE_, or just to make them as struct fields in the existing cmd configuration and handle them just as any other variables.